test/localtest.sh

```diff
@@ -6,14 +6,23 @@ if [ "${EXPECTED_FLAVOR}" != "datadog-agent" ] && echo "${SCRIPT}" | grep "agent
     exit 0
 fi
 
-$SCRIPT
+cp $SCRIPT /tmp/script.sh
+if [ "$DD_APM_INSTRUMENTATION_ENABLED" == "all" ] || [ "$DD_APM_INSTRUMENTATION_ENABLED" == "docker" ] || echo "${SCRIPT}" | grep "docker_injection.sh$" > /dev/null; then
+    # fake presence of docker and make sure the script doesn't try to restart it
+    mkdir /etc/docker
+    sed -i "s|dd-container-install --no-agent-restart|dd-container-install --no-agent-restart --no-docker-reload|" /tmp/script.sh
+fi
+/tmp/script.sh
 
 INSTALLED_VERSION=
 RESULT=0
 EXPECTED_MAJOR_VERSION=6
 if echo "${SCRIPT}" | grep "agent7.sh$" >/dev/null || [ "${EXPECTED_FLAVOR}" != "datadog-agent" ] ; then
     EXPECTED_MAJOR_VERSION=7
 fi
+if echo "${SCRIPT}" | grep "docker_injection.sh$" > /dev/null; then
```

Comment:
Nice to have: we're repeating this conditional a few times, would be nice to set it once and reuse.

Reply:
You're right, I'll fix that.